### PR TITLE
Added binarysearch in python2

### LINKED ISSUE #377: BinarySearch in Python2
This is an open issue for binary search algorithm implementation in Python2. If you wish to contribute, just make some noise and we will assign you.

### PR DESCRIPTION
**Pull request by avsd47**
**Hi!**  
Thanks for considering contributing to this ever-growing list of algorithm and data structure implementations.
Your contribution is valuable. Use the follow template for creating Pull Requests
---

**Fixes #377 **

By submitting this pull request I confirm I've read and complied with the below declarations.

- ✔   I have read the [Contribution guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md) 
and I am confident that my PR reflects them.

-  ✔ I have followed the [coding guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#cs) 
for this project.

- ✔ My code follows the [skeleton code structure](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#sample).

- ✔ This pull request has a descriptive title. 
For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.

- ✔ This pull request shall only be merged once the build passes.

- ✔ I have mentioned the issue number correctly (with hyper-link) in this pull request description.

### CODE REVIEW

**Comment by Danushka96** [comment]
Hi @avsd47 
I can't see any file changes here,

**Review by RavinduSachintha — CHANGES_REQUESTED** [review]
there is a little issue. just add a new line add the end of the file. It is about best practices, nothing else. :+1:

File: algorithms/ar-bsrh/python2/binary_search.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  low=0
+  high=len(arr)-1
+  mid=0
+  while (low<=high): 
+    mid=(low+high)//2
+    if (arr[mid]<x):
+      low=mid+1
+    elif (arr[mid]>x):
+      high=mid-1
+    else:
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:11]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277716)

File: algorithms/ar-bsrh/python2/binary_search.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+def binarysearch(arr,x):
+  low=0
+  high=len(arr)-1
+  mid=0
+  while (low<=high):
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:5]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277716)

File: algorithms/ar-bsrh/python2/binary_search.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  while (low<=high): 
+    mid=(low+high)//2
+    if (arr[mid]<x):
+      low=mid+1
+    elif (arr[mid]>x):
+      high=mid-1
+    else: 
+      return mid
+  return -1
+
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:14]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277716)

File: algorithms/ar-bsrh/python2/binary_search.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+      high=mid-1
+    else: 
+      return mid
+  return -1
+  
+def main():
+  arr = [1,2,3,4,5,6]
+  index = binarysearch(arr,5)
+  if (index==-1):
+    print "The element is not found!"
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:19]: Codacy found an issue: [invalid syntax (F999)](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277716)

File: algorithms/ar-bsrh/python2/binary_search.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  low=0
+  high=len(arr)-1
+  mid=0
+  while(low<=high): 
+    mid=(low+high)//2
+    if(arr[mid]<x):
+      low=mid+1
+    elif(arr[mid]>x):
+      high=mid-1
+    else:
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:11]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277716)

File: algorithms/ar-bsrh/python2/binary_search.py [hunk 6]
```diff
@@ -0,0 +1,27 @@
+def binarysearch(arr,x):
+  low=0
+  high=len(arr)-1
+  mid=0
+  while(low<=high):
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:5]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277716)

File: algorithms/ar-bsrh/python2/binary_search.py [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  while(low<=high): 
+    mid=(low+high)//2
+    if(arr[mid]<x):
+      low=mid+1
+    elif(arr[mid]>x):
+      high=mid-1
+    else: 
+      return mid
+  return -1
+
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:14]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277716)

File: algorithms/ar-bsrh/python2/binary_search.py [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+      high=mid-1
+    else: 
+      return mid
+  return -1
+  
+def main():
+  arr = [1,2,3,4,5,6]
+  index = binarysearch(arr,5)
+  if(index==-1):
+    print "The element is not found!"
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:19]: Codacy found an issue: [invalid syntax (F999)](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277716)

File: algorithms/ar-bsrh/python2/binary_search.py (hunk 6 above)
Comment (RavinduSachintha) [reply algorithms/ar-bsrh/python2/binary_search.py:5]: this is happening when you have added unwanted blank spaces at the end of the code lines. just remove them and the issue will be gone.

File: algorithms/ar-bsrh/python2/binary_search.py (hunk 5 above)
Comment (RavinduSachintha) [reply algorithms/ar-bsrh/python2/binary_search.py:11]: this is also the same mistake.

File: algorithms/ar-bsrh/python2/binary_search.py (hunk 7 above)
Comment (RavinduSachintha) [reply algorithms/ar-bsrh/python2/binary_search.py:14]: same

File: algorithms/ar-bsrh/python2/binary_search.py (hunk 8 above)
Comment (RavinduSachintha) [reply algorithms/ar-bsrh/python2/binary_search.py:19]: you should use single quotations (not double quotations) with the print command. it is just a guidance rule.

**Comment by avsd47** [comment]
I edited it could you please review again and let me know if any issue persists

**Comment by RavinduSachintha** [comment]
> I edited it could you please review again and let me know if any issue persists

Your changes didn't push to the PR yet.

File: algorithms/ar-bsrh/python2/binary_search.py [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+  low=0
+  high=len(arr)-1
+  mid=0
+  while(low<=high): 
+    mid=(low+high)//2
+    if(arr[mid]<x):
+      low=mid+1
+    elif(arr[mid]>x):
+      high=mid-1
+    else:
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:11]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277716)

File: algorithms/ar-bsrh/python2/binary_search.py [hunk 10]
```diff
@@ -0,0 +1,25 @@
+def binarysearch(arr,x):
+  low=0
+  high=len(arr)-1
+  mid=0
+  while(low<=high):
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:5]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277716)

File: algorithms/ar-bsrh/python2/binary_search.py [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+  mid=0
+  while(low<=high): 
+    mid=(low+high)//2
+    if(arr[mid]<x):
+      low=mid+1
+    elif(arr[mid]>x):
+      high=mid-1
+    else: 
+      return mid
+  return -1
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:13]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277716)

File: algorithms/ar-bsrh/python2/binary_search.py [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+    elif(arr[mid]>x):
+      high=mid-1
+    else: 
+      return mid
+  return -1 
+def main():
+  arr = [1,2,3,4,5,6]
+  index = binarysearch(arr,5)
+  if(index==-1):
+    print 'The element is not found!'
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:18]: Codacy found an issue: [invalid syntax (F999)](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277716)

**Comment by avsd47** [comment]
Thank you for your patience i=now igot clear understanding with the way github is working thanks a lot for being patient and i think now it's fine

File: algorithms/ar-bsrh/python2/binary_search.py (hunk 10 above)
Comment (RavinduSachintha) [reply algorithms/ar-bsrh/python2/binary_search.py:5]: there is an unwanted white space at the end of line 5.

**Comment by RavinduSachintha** [comment]
> Thank you for your patiencenow i got clear understanding with the way github is working thanks a lot for being patient and i think now it's fine

nothing to worry about that. but still, you see some minor issues. right? until you fix them we can not merge to the master branch.

File: algorithms/ar-bsrh/python2/binary_search.py [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+  low=0
+  high=len(arr)-1
+  mid=0
+  while(low<=high): 
+    mid=(low+high)//2
+    if(arr[mid]<x):
+      low=mid+1
+    elif(arr[mid]>x):
+      high=mid-1
+    else:
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:11]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277716)

File: algorithms/ar-bsrh/python2/binary_search.py [hunk 14]
```diff
@@ -0,0 +1,26 @@
+def binarysearch(arr,x):
+  low=0
+  high=len(arr)-1
+  mid=0
+  while(low<=high):
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:5]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277716)

File: algorithms/ar-bsrh/python2/binary_search.py [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+      high=mid-1
+    else: 
+      return mid
+  return -1 
+
+def main():
+  arr = [1,2,3,4,5,6]
+  index = binarysearch(arr,5)
+  if(index==-1):
+    print 'The element is not found!'
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:19]: Codacy found an issue: [invalid syntax (F999)](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277716)

File: algorithms/ar-bsrh/python2/binary_search.py [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+  else:
+    print 'Element is found at index:',index
+
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:22]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277716)

File: algorithms/ar-bsrh/python2/binary_search.py [hunk 17]
```diff
@@ -0,0 +1,26 @@
...
+  mid=0
+  while(low<=high): 
+    mid=(low+high)//2
+    if(arr[mid]<x):
+      low=mid+1
+    elif(arr[mid]>x):
+      high=mid-1
+    else: 
+      return mid
+  return -1
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:13]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277716)

File: algorithms/ar-bsrh/python2/binary_search.py (hunk 14 above)
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:5]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277716)

File: algorithms/ar-bsrh/python2/binary_search.py (hunk 17 above)
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:13]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277716)

File: algorithms/ar-bsrh/python2/binary_search.py (hunk 13 above)
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:11]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277716)

File: algorithms/ar-bsrh/python2/binary_search.py (hunk 16 above)
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:22]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277716)

File: algorithms/ar-bsrh/python2/binary_search.py (hunk 15 above)
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:19]: Codacy found an issue: [invalid syntax (F999)](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277716)

File: algorithms/ar-bsrh/python2/binary_search.py (hunk 17 above)
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:13]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277716)

File: algorithms/ar-bsrh/python2/binary_search.py (hunk 13 above)
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:11]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277716)

File: algorithms/ar-bsrh/python2/binary_search.py (hunk 14 above)
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:5]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277716)

File: algorithms/ar-bsrh/python2/binary_search.py (hunk 16 above)
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/python2/binary_search.py:22]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6277716)

**Comment by RavinduSachintha** [comment]
@avsd47 If you wish to continue this contribution, you have to fix just a few minor issues here. Otherwise, we will have to close this PR without merging.

**Comment by avsd47** [comment]
I was a bit busy yesterday now i am looking into it,i will fix this soon

**Comment by avsd47** [comment]
Could you please review again i made some changes

**Review by RavinduSachintha — APPROVED** [review]
Nice work @avsd47 
It would be great if you can add this contribution to main README file too. You just need to copy-paste the phrase for "correct sign" to the right place in the algorithm chart. No need to make another PR. just change the README file and push it.

**Comment by avsd47** [comment]
Could you please review the changes i have added myself as contributor and also added the checkmark

File: README.md [hunk 18]
```diff
@@ -161,6 +161,7 @@ Only project maintainers should merge a PR. Other members can add their reviews
...
     <td align="center"><a href="https://github.com/NidhiNivedita6"><img src="https://avatars2.githubusercontent.com/u/56748276?v=4" width="100px;" alt=""/><br /><sub><b>Nidhi Nivedita</b></sub></a><br /><a href="https://github.com/FOSS-UCSC/FOSSALGO/commits?author=NidhiNivedita6" title="Code">💻</a></td>
     <td align="center"><a href="https://github.com/Aaditya1978"><img src="https://avatars3.githubusercontent.com/u/56872630?v=4" width="100px;" alt=""/><br /><sub><b>Aaditya1978</b></sub></a><br /><a href="https://github.com/FOSS-UCSC/FOSSALGO/commits?author=Aaditya1978" title="Code">💻</a></td>
+    <td align="center"><a href="https://github.com/avsd47"><img src="https://avatars0.githubusercontent.com/u/47289942?s=400&u=29eb591b7decff18890e929d4c0593dd86f7236a&v=4" width="100px;" alt=""/><br /><sub><b>avsd47</b></sub></a><br /><a href="https://github.com/FOSS-UCSC/FOSSALGO/commits?author=avsd47" title="Code">💻</a></td>
```
Comment (RavinduSachintha) [inline README.md:164]: please revert this change. This part is handling by an automated robot.

File: README.md [hunk 19]
```diff
@@ -22,7 +22,7 @@ Community maintained list of Algorithm and Data Structure implementations.
 | Algorithm                        | C                  | CPP                | Java               | Python2            | Python3            | Golang | JavaScript         | C#    | TypeScript |
 | :---                             | :---:              | :---:              | :---:              | :---:              | :---               | :---:  | :---:              | :---: | :---:      |
 | Bin Sort                         |                    |                    |                    |                    |                    |        |                    |       |            |
-| Binary Search                    | :white_check_mark: | :white_check_mark: | :white_check_mark: |                    | :white_check_mark: |        | :white_check_mark: |       |            |
+| Binary Search                    | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: |        | :white_check_mark: |       |            |
```
Comment (RavinduSachintha) [inline README.md:25]: :+1:

**Comment by avsd47** [comment]
I reverted it could you please review it again

**Comment by RavinduSachintha** [comment]
@all-contributors please add @avsd47  for code and doc
@avsd47 Thank you for your contribution. 
Good luck! :+1: 
Make sure to star this repository. :wink:

**Pull request merged**